Assets/Scripts/EOSAchievementManager.cs

```diff
@@ -205,23 +119,27 @@ public void OnConnectLogin(Epic.OnlineServices.Connect.LoginCallbackInfo loginCa
         /// to any player for whom that information has been requested
         /// previously.
         /// </summary>
-        public void Refresh()
+        public async override void Refresh()
         {
             ProductUserId productUserId = EOSManager.Instance.GetProductUserId();
-            QueryAchievements(productUserId, (ref OnQueryDefinitionsCompleteCallbackInfo defQueryData) =>
-            {
-                _achievements = GetCachedAchievements();
-                foreach (var userId in _playerStats.Keys)
+            _achievements = await QueryAchievementsAsync(productUserId);
+            
+                foreach (var userId in _playerAchievements.Keys)
                 {
-                    RefreshPlayerStatsAndAchievements(userId);
+                    RefreshPlayerAchievementsAsync(userId);
```

Comment:
Is there a reason we're not awaiting a Task for each of these? This _looks_ like something that could create race conditions or inconsistent results.

Reply:
I have updated the implementation to utilize `await` for each of these calls.

Reply:
`await Task.WhenAll` is a great use of the C# API here, I had just thought about awaiting for each but this is much better and more thread safe.